hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/QuotaCache.java

```diff
@@ -70,10 +69,10 @@ public class QuotaCache implements Stoppable {
   private final Object initializerLock = new Object();
   private volatile boolean initialized = false;
 
-  private volatile Map<String, QuotaState> namespaceQuotaCache = new HashMap<>();
-  private volatile Map<TableName, QuotaState> tableQuotaCache = new HashMap<>();
-  private volatile Map<String, UserQuotaState> userQuotaCache = new HashMap<>();
-  private volatile Map<String, QuotaState> regionServerQuotaCache = new HashMap<>();
+  private volatile Map<String, QuotaState> namespaceQuotaCache = new ConcurrentHashMap<>();
```

Comment:
Do these need to be concurrent if we're also synchronizing on `this`?

Reply:
Yes, because changes to their contents are made visible while the synchronized block is in progress